webui: introduce themes

fbergkemper opened the pull request:
Introduces two themes called `default` and `sunflower`. The `sunflower` theme is default for now.

### Thank you for contributing to the Bareos Project!

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)


### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems

sebastianlederer commented on docs/manuals/source/IntroductionAndTutorial/InstallingBareosWebui.rst:
```diff
@@ -383,6 +383,14 @@ Since :sinceVersion:`16.2.2: /etc/bareos-webui/configuration.ini` you are able t
+   ;------------------------------------------------------------------------------
+   [theme]
+   ; Possible values: bareos, ukraine
+   ; Default: bareos
```
The default at the moment is actually "ukraine", but that will be changed back to "bareos" at some point in the future.

fbergkemper replied:
> What is the slate theme about?

It's mentioned in the PR description.

> You want to mention it also?

No, it's experimental and working but not finished yet.

fbergkemper replied:
> As a lot of files in the different theme directories are identical: how about linking the identical files to the default (bareos) theme?

No, better keep them separate. Something might change in future in one or the other. This is just the beginning of having themes.

> Probable also rename the bareos theme to `default`?

Yeah, we could do that.

fbergkemper replied:
> The documentation should represent the current status. Changing the default will require changing the code. The documentation can than also be changed.

Will change that in https://github.com/bareos/bareos/pull/1281/commits/0151964fd574986a94de2525d1bb2aa623be2929 so we can simply git revert to the actual default at some point in time.

fbergkemper replied:
> > The documentation should represent the current status. Changing the default will require changing the code. The documentation can than also be changed.
> 
> Will change that in [0151964](https://github.com/bareos/bareos/commit/0151964fd574986a94de2525d1bb2aa623be2929) so we can simply git revert to the actual default at some point in time.

Changed here: https://github.com/bareos/bareos/pull/1281/commits/c06180bcc0d1ad5fd80e84b8dee489679375d0cc

fbergkemper replied:
Can we resolve this conversation?

fbergkemper commented on webui/install/configuration.ini.in:
```diff
@@ -49,3 +49,11 @@ labelpooltype=scratch
+[theme]
+; Possible values: bareos, ukraine
+; Default: ukraine
+name=ukraine
```
> Please rename the theme `ukraine` to `sunflower_and_heaven ` (or something similar).

Will do so.

fbergkemper replied:
> > Please rename the theme `ukraine` to `sunflower_and_heaven ` (or something similar).
> 
> Will do so.

Renamed to sunflower to keep it short.

fbergkemper replied:
Can we resolve this conversation?

sebastianlederer reviewed (changes requested):
The file bareos-logo.png does not seem to be used at all. In my test install, I deleted these files without any noticable effect:

        deleted:    public/img/bareos-logo.png
        deleted:    public/themes/default/img/bareos-logo.png
        deleted:    public/themes/slate/img/bareos-logo.png
        deleted:    public/themes/sunflower/img/bareos-logo.png

So it is likely that these files can be removed entirely.

The other redundant PNG files and font files are probably unavoidable without much more changes in the PHP code and in the bootstrap CSS.

fbergkemper commented:
> The file bareos-logo.png does not seem to be used at all. In my test install, I deleted these files without any noticable effect:
> 
> ```
>     deleted:    public/img/bareos-logo.png
>     deleted:    public/themes/default/img/bareos-logo.png
>     deleted:    public/themes/slate/img/bareos-logo.png
>     deleted:    public/themes/sunflower/img/bareos-logo.png
> ```
> 
> So it is likely that these files can be removed entirely.
> 
> The other redundant PNG files and font files are probably unavoidable without much more changes in the PHP code and in the bootstrap CSS.

I'd really like to avoid removing original source image material from source tree even if it's not actively used in the application currently.

If a theme doesn't and probably never will use the mentioned one above, we could remove it, I agree.

But let's keep and not touch originals in the default image folder (public/img/).

Our README.md for example references an image located in here as well,
another reason I kept the folder and images for now.

If we want to reduce application size, build and packaging should handle this.

We could move the img folder one directory up to place it in webui root folder and have the source image material available there, I guess. That might already exclude it from being packaged, but I have to check that out.

In addition this change would require to adjust the image reference in our https://github.com/bareos/bareos/blob/master/README.md .

fbergkemper commented:
Removed unused image material from themes except one source image for reference in the default theme folder I'd like to keep for now.

Also adjusted the Bareos logo include in our README.md in a separate commit due to the recent changes.

joergsteffens commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# <img src="https://raw.githubusercontent.com/bareos/bareos/master/webui/public/img/bareos.png" alt="Bareos" />
+# <img src="https://raw.githubusercontent.com/bareos/bareos/master/webui/public/themes/default/img/bareos.png" alt="Bareos" />
```
Currently we show the logo from the sunflower theme. did you change this intentionally? If yes, how about using thew logo from https://github.com/bareos/bareos/raw/master/docs/manuals/images/bareos-full-logo.png ?

fbergkemper replied:
Well, currently it's shown there because we simply have overwritten the original logos in the referenced webui folder. So the replacement in the README.md wasn't really on purpose back in the days, I think.

fbergkemper replied:
Going to replace the logo in README.md with the one located in docs.

fbergkemper replied:
Image source replaced, see https://github.com/bareos/bareos/pull/1281/commits/8970e82ef1270b25872431d73ee08cab33dece63 .

fbergkemper replied:
I think we can resolve this conversation, do you agree?

joergsteffens commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# <img src="https://raw.githubusercontent.com/bareos/bareos/master/webui/public/img/bareos.png" alt="Bareos" />
+# <img src="https://raw.githubusercontent.com/bareos/bareos/master/webui/public/themes/default/img/bareos.png" alt="Bareos" />
 
 [![Build Status](https://travis-ci.com/bareos/bareos.svg?branch=master)](https://travis-ci.com/bareos/bareos/branches)
```
```suggestion
```
Travis is no longer used.

fbergkemper replied:
Removed, see https://github.com/bareos/bareos/pull/1281/commits/8970e82ef1270b25872431d73ee08cab33dece63 .

fbergkemper replied:
This conversation can be closed, I think.

fbergkemper commented:
Removed the experimental 'slate' theme and reduced the left themes to its minimum.

fbergkemper commented:
Updated PR title and description.

fbergkemper commented:
CHANGELOG.md needs to be adjusted.

fbergkemper commented:
We should also squash commits before merge for a clean commit history. I kept it due to the review process for now as it is.

joergsteffens commented:
It might be good idea, to refer to the source of the sunflower picture, at least in the git commit message.
